src/js/modules/supervisors/FileManager.ts

```diff
@@ -395,6 +401,37 @@ class FileManager {
       this.removeHandleFromFilePath(filePath, repository);
     });
   }
+
+  /**
+   * Lazily load the ManagementClient
+   * Retires and sleeps 1s between configurable max number of attempts
+   *
+   * @param retries - Max number of connection attempts
+   * @return Promise with management client on success or error on failure
+   */
+  private getClient(): Promise<ManagementClient> {
+    if (!this.managementClient) {
+      return ManagementClient.create(43118)
+        .then((client) => {
+          logger.info("Succeeded in establishing a connection to redpanda");
+          this.managementClient = client;
+          return client;
+        })
+        .catch((err) => {
+          logger.warn(
+            `Failed to connect to redpanda, retrying again, reason: ${err.message}`
+          );
+          return new Promise((resolve, reject) => {
```

Comment:
we should add some limit for it, like 5 tries and fail, what do you think?

Reply:
@senior7515 mentioned during yesterdays standup that we should just never stop trying, makes sense in case redpanda comes back up after a while, things will continue as normal

Reply:
Got it